examples/adrv9009_som_multi.py

```diff
@@ -0,0 +1,169 @@
+# type: ignore
+
+import csv
+import os
+import time
+
+import adi
+import matplotlib.pyplot as plt
+import numpy as np
+import scipy.io as sio
+from scipy import signal
+
+
+def measure_phase_and_delay(chan0, chan1, window=None):
+    assert len(chan0) == len(chan1)
+    if window == None:
+        window = len(chan0)
+    phases = []
+    delays = []
+    indx = 0
+    sections = len(chan0) // window
+    for sec in range(sections):
+        chan0_tmp = chan0[indx : indx + window]
+        chan1_tmp = chan1[indx : indx + window]
+        indx = indx + window + 1
+        cor = np.correlate(chan0_tmp, chan1_tmp, "full")
+        # plt.plot(np.real(cor))
+        # plt.plot(np.imag(cor))
+        # plt.plot(np.abs(cor))
+        # plt.show()
+        i = np.argmax(np.abs(cor))
+        m = cor[i]
+        sample_delay = len(chan0_tmp) - i - 1
+        phases.append(np.angle(m) * 180 / np.pi)
+        delays.append(sample_delay)
+    return (np.mean(phases), np.mean(delays))
+
+
+def measure_phase(chan0, chan1):
+    assert len(chan0) == len(chan1)
+    errorV = np.angle(chan0 * np.conj(chan1)) * 180 / np.pi
+    error = np.mean(errorV)
+    return error
+
+
+buff_size = 2 ** 14
+
+# Create radio
+primary = "ip:10.44.3.39"
+secondary = "ip:10.44.3.50"
+
+lo_freq = 1000000000
+dds_freq = 7000000
+
+primary_jesd = adi.jesd(primary)
+secondary_jesd = adi.jesd(secondary)
+
+print("--Connecting to devices")
+multi = adi.adrv9009_zu11eg_multi(
+    primary, [secondary], primary_jesd, [secondary_jesd], fmcomms8=True
+)
+
+multi._dma_show_arming = False
+multi._jesd_show_status = True
+multi._jesd_fsm_show_status = True
+multi._clk_chip_show_cap_bank_sel = True
+multi._resync_tx = True
+multi.rx_buffer_size = 2 ** 10
+
+multi.primary._clock_chip_ext.reg_write(0xCF, 1)
+multi.primary._clock_chip_ext.reg_write(0xE3, 1)
```

Comment:
Can we add a method for this?

Reply:
Totally agree - ideally we have a hmc7044 class.
But having a method with takes the output channel as parameter and a delay setting is definitely good to have.

Reply:
Yes, a class for 7044 would be nice. Would be want to put these functions in the driver first though? I don't know if we would want a class that just has a bunch of register writes in it.

Reply:
> Would be want to put these functions in the driver first though? 

Well they are already. But the digital delay is expressed as phase in RAD and changes by frequency. Not something that you want if you just want to specify a delay.
Both digital and analog delay can be set via devicetree for each channel.
The function here is to simply determine values which work and can be put into the devicetree afterwards.